eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -325,7 +325,7 @@ object Graph {
         val riskCost = totalAmount.toLong * totalCltv.toInt * heuristicsConstants.lockedFundsRisk
         // If the edge was added by the invoice, it is assumed that it can route the payment.
         // If we know the balance of the channel, then we will check separately that it can relay the payment.
-        val successProbability = if (edge.update.chainHash == ByteVector32.Zeroes || edge.balance_opt.nonEmpty) 1.0 else 1.0 - totalAmount.toLong.toDouble / edge.capacity.toMilliSatoshi.toLong.toDouble
+        val successProbability = if (edge.update.chainHash == ByteVector32.Zeroes || edge.balance_opt.nonEmpty) 1.0 else 1.0 - prev.amount.toLong.toDouble / edge.capacity.toMilliSatoshi.toLong.toDouble
```

Comment:
It feels to me that we should still take `totalAmount` into account since that's what will flow through this channel, isn't it?
But we should do a `.max(0)` to ensure it cannot be negative, WDYT? 

Otherwise a future reader of that code may think "this is a mistake, we should use `totalAmount`" and would "fix" this without realizing that they're re-introducing a bug.

Reply:
No, only `prev.amount` will flow through the channel.

Reply:
Don't you think it's safer to add a `.max(0)` anyway though?

Reply:
If the probability is negative, it means we have a bug somewhere. Adding `.max(0)` will only make sure that we never discover that bug.

Reply:
Fair enough, let's hope we catch those bugs before production though

Reply:
Or better, we can add a `require`, this way if it happens it crashes the actor and tells us exactly what happened, instead of causing all kinds of unexpected behavior?

Reply:
I could add an assert if you think there is a risk. That way we would get something logged, similar to what I did with `InfiniteLoop`.

Reply:
It's your call, whatever you think is best here.